agent/lib/kontena/workers/node_info_worker.rb

```diff
@@ -23,37 +24,59 @@ def initialize(autostart = true)
       @statsd = nil
       @stats_since = Time.now
       @container_seconds = 0
-      subscribe('websocket:connected', :on_websocket_connected)
-      subscribe('agent:node_info', :on_node_info)
       subscribe('container:event', :on_container_event)
       info 'initialized'
       async.start if autostart
     end
 
     def start
       loop do
+        node = self.node
+        on_node_info(node)
         sleep PUBLISH_INTERVAL
         self.publish_node_info
         self.publish_node_stats
       end
     end
 
-    # @param [String] topic
-    # @param [Hash] data
-    def on_websocket_connected(topic, data)
-      self.publish_node_info
-      self.publish_node_stats
+    # @return [Node, NilClass]
+    def fetch_node
+      request = rpc_client.request('/nodes/get', docker_info['ID'])
+      info = request.value
+      if info
+        node = Node.new(info)
+        node
+      end
+    rescue => exc
+      error exc.message
+      nil
     end
 
-    # @param [String] topic
-    # @param [Hash] info
-    def on_node_info(topic, info)
-      statsd_conf = info.dig('grid', 'stats', 'statsd')
-      if statsd_conf
+    # @return [Node]
+    def node
+      if @node.nil?
+        while @node.nil?
+          @node = fetch_node
```

Comment:
Yes, just wondering should we just fetch/update node info on every iteration or not.